KIPs/kip-71.md

```diff
@@ -0,0 +1,385 @@
+				signer_address = signer_address,
+				signer_nonce = transaction.nonce,
+				max_fee_per_gas = transaction.gas_price,
+				gas_limit = transaction.gas,
+				to = transaction.to,
+				value = transaction.value,
+				data = transaction.input,
+				fee_payer_address = transaction.fee_payer,
+				fee_ratio = None
+			)
+		
+		elif isinstance(transaction, TxTypeFeeDelegatedSmartContractExecutionWithRatio):
+			return NormalizedTransaction(
+				signer_address = signer_address,
+				signer_nonce = transaction.nonce,
+				max_fee_per_gas = transaction.gas_price,
+				gas_limit = transaction.gas,
+				to = transaction.to,
+				value = transaction.value,
+				data = transaction.input,
+				fee_payer_address = transaction.fee_payer,
+				fee_ratio = transaction.fee_ratio
+			)
+		elif isinstance(transaction, TxTypeEthereumAccessList):
+			return NormalizedTransaction(
+				signer_address = signer_address,
+				signer_nonce = transaction.nonce,
+				max_fee_per_gas = transaction.gas_price,
+				gas_limit = transaction.gas,
+				to = transaction.to,
+				value = transaction.value,
+				data = transaction.data,
+				fee_payer_address = None,
+				fee_ratio = None
+			)
+		elif isinstance(transaction, TxTypeEthereumDynamicFee):
+			return NormalizedTransaction(
+				signer_address = signer_address,
+				signer_nonce = transaction.nonce,
+				max_fee_per_gas = transaction.gas_fee_cap,
+				gas_limit = transaction.gas,
+				to = transaction.to,
+				value = transaction.value,
+				data = transaction.data,
+				fee_payer_address = None,
+				fee_ratio = None
+			)
+		else:
+			raise Exception('invalid transaction: unexpected number of items')
+
+	@abstractmethod
+	def parent(self, block: Block) -> Block: pass
+
+	@abstractmethod
+	def block_hash(self, block: Block) -> int: pass
+
+	@abstractmethod
+	def transactions(self, block: Block) -> Sequence[Transaction]: pass
+
+	# effective_gas_price is the value returned by the GASPRICE (0x3a) opcode
+	@abstractmethod
+	def execute_transaction(self, transaction: NormalizedTransaction, effective_gas_price: int) -> int: pass
+
+	@abstractmethod
+	def validate_and_recover_signer_address(self, transaction: Transaction) -> int: pass
+
+	@abstractmethod
+	def account(self, address: int) -> Account: pass
+```
+
+
+## Expected Effect
+
+The proposed dynamic gas price mechanism is expected to solve the following problems.
+* Reduce transaction process delay and storage overcapacity
+* Flexible gas price control depending on the network status
+* Introduction of a burn mechanism
+
+But it may give rise to the following changes:
+* Edit gas price-related code of the ecosystem wallet and other tools
+* Lower predictability of gas price
+
+## Backwards Compatibility
+
+* All previous transaction types will remain the same.
+* Setting fixed gas price to send transactions will still work and be included in blocks, but those transactions will be not included in the block if a `base_fee` is higher than the gas price.
+
```

Comment:
I suggest to replace `base_fee` to `base_fee_per_gas` to avoid confusion.
Furthermore I'd like to emphasize `base_fee_per_gas` in txPool and `gas_price` in tx.

Reply:
Everything else LGTM.